cmd/infra/aws/create_cli_role.go

```diff
@@ -0,0 +1,246 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/spf13/cobra"
+
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	"github.com/openshift/hypershift/cmd/log"
+)
+
+const (
```

Comment:
big-picture: how did you get this list of actions? Can we make it derivative of other roles?

Reply:
I basically went through the cli code and transformed each AWS API call to a permission, tested it -> got an unauthorized error -> added the missing perm and so on :)

Reply:
> Can we make it derivative of other roles?

what other roles?

Reply:
In standalone OCP, is it a cluster operator making these calls, or is it e.g. `ccoctl`?

Reply:
this role is meant to be used by our cli to provision infra for hostedCluster instead of using aws creds directly. This will not be passed to or used by any openshift operators/tools.

This is just the first part, Patryk is working to modify the cli code to use `AssumeRole` with this role.

Reply:
Right, I understand that - I was wondering if any of this was already captured in the standalone bits or not. We have a lot of code in `hypershift create infra` and `hypershift create iam` that is 100% supported and written in e.g. the `ccoctl` binary, and it seems like a reasonable idea to DRY things out at some point so we're not writing and maintaining our own copy. If, for instance, these permissions were already captured in some place in OCP core, we could import them as opposed to listing them ourselves. If not, then no worries.

Reply:
I see, yeah makes sense if/when we import/vendor those infra creation bits from OCP at some point.

But, I wouldn't look to derive those perms today as long as those functionalities are implemented directly in our code.